relevanceai/http_client.py

```diff
@@ -100,3 +102,48 @@ def check_auth(self):
             return response.status_code == 200    
         except:
             raise Exception("Invalid auth details.")
+
+    def kmeans_cluster(self,
+        dataset_id: str,
+        vector_fields: list,
+        filters: List = [],
+        k: Union[None, int] = 10,
+        init: str = "k-means++",
+        n_init: int = 10,
+        max_iter: int = 300,
+        tol: float = 1e-4,
+        verbose: bool = True,
+        random_state = None,
+        copy_x: bool = True,
+        algorithm: str ="auto",
+        alias: str = "default"
+    ):
+        # load the documents
+        docs = self.get_all_documents(dataset_id=dataset_id, filters=filters, select_fields=vector_fields)
+
+        # Cluster
+        clusterer = KMeans(
+            k = k,
+            init = init,
+            n_init = n_init,
+            max_iter= max_iter,
+            tol= tol,
+            verbose = verbose,
+            random_state = random_state,
+            copy_x = copy_x,
+            algorithm = algorithm)
+        clustered_docs = clusterer.fit_documents(vector_fields, docs, return_only_clusters=True)
+
+        # Write back the results
+        for i, doc in enumerate(docs):
```

Comment:
You will want to use: 
`self._set_field_across_documents()` 

See here: 
https://github.com/RelevanceAI/python-doc-utils/blob/main/doc_utils/write_utils.py#L72